kafka/producer.go

```diff
@@ -17,4 +17,164 @@
 
 package kafka
 
-// TODO
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+
+	"go.uber.org/zap"
+
+	"github.com/twmb/franz-go/pkg/kgo"
+	"github.com/twmb/franz-go/plugin/kzap"
+
+	"github.com/elastic/apm-data/model"
+	"github.com/elastic/apm-queue/queuecontext"
+)
+
+type RecordMutator func(model.APMEvent, *kgo.Record) error
+
+// ProducerConfig holds configuration for publishing events to Kafka.
+type ProducerConfig struct {
+	// Broker holds the (host:port) address of the Kafka broker to which
+	// events should be published.
+	Broker string
+
+	// ClientID to use when connecting to Kafka. This is used for logging
+	// and client identification purposes.
+	ClientID string
+	// Version is the software version to use in the Kafka client. This is
+	// useful since it shows up in Kafka metrics and logs.
+	Version string
+
+	// Logger is used for logging producer errors.
+	Logger *zap.Logger
+
+	// Sync can be used to indicate whether production should be synchronous.
+	Sync bool
+
+	Mutators []RecordMutator
```

Comment:
Can you add a GoDoc description for this field?

Reply:
Good catch! Updated!